### ISSUE TITLE
Enhancement Request: Use $result$ tokens in alert definition and ignore _fields in results

### ISSUE THREAD

**Issue by afxmac** [alertmanager/alert_manager#263]
When defining email alerts in Splunk, one can use $result.fieldname$ for alert titles, that would be nice in alert manager too.
And if a field starts with _ this is not transported into the alert, with alert-manager I still see _ fields.

What I typically use in email alerts is something like $result._field$ for the title and the _field is not shown in the alert result, it has been constructed especially for the title and is then ignored thanks the _ at the beginning of the name.

thx
afx

**Comment by my2ndhead** [comment]
Can you please clarify? You can set the e-mail subject dynamically in Alert Manager 
![image](https://user-images.githubusercontent.com/10104343/90133365-9e969b00-dd6f-11ea-944a-6a7c7f5bfbb8.png)

**Comment by afxmac** [comment]
Yes, the e-mail subject works fine, as I clarified above in the edit. But, the trick to hide the fields used for the subject from the alert  by prefixing fields with "_" does not work.
Sorry for wording this in a misleading form.

**Comment by my2ndhead** [comment]
Which fields are in the body is done using templates.

https://github.com/alertmanager/alert_manager/blob/0d72758c9bd2033297c66eb2aac776058b5471e9/src/default/templates/default.html#L57

Does creating your own templates solve the problem?

**Comment by afxmac** [comment]
Hmm, my JavaScript skills are rather minimal...
How would I exclude fields that have a "_" at the start?

**Comment by my2ndhead** [comment]
It's a Python Django Template :-)

You will need to add a filter to nor render a column, when the field starts with underscore.

https://docs.djangoproject.com/en/3.1/ref/templates/builtins/

**Comment by afxmac** [comment]
Here goes my weekend...
thx
afx
